src/consensus/read_validator_test.rs

```diff
@@ -143,4 +144,58 @@ mod tests {
             }
         );
     }
+
+    #[tokio::test]
+    async fn test_ignore_no_signatures() {
+        let (mut proposer_engine, _read_node_engine, mut read_validator, proposer_keypair) =
+            setup(0).await;
+        let mut shard_chunk = commit_shard_chunk(&mut proposer_engine, &proposer_keypair).await;
+        // Remove signatures from chunk
+        shard_chunk.commits = Some(Commits {
+            height: shard_chunk.header.as_ref().unwrap().height,
+            round: 0,
+            value: Some(ShardHash {
+                shard_index: shard_chunk
+                    .header
+                    .as_ref()
+                    .unwrap()
+                    .height
+                    .unwrap()
+                    .shard_index,
+                hash: shard_chunk.hash.clone(),
+            }),
+            signatures: vec![],
+        });
+        let num_processed = process_decided_value(&mut read_validator, &shard_chunk).await;
+        assert_eq!(num_processed, 0);
+    }
+
+    #[tokio::test]
+    async fn test_ignore_invalid_signatures() {
+        let (mut proposer_engine, _read_node_engine, mut read_validator, proposer_keypair) =
+            setup(0).await;
+        let mut shard_chunk = commit_shard_chunk(&mut proposer_engine, &proposer_keypair).await;
```

Comment:
Might be a better test to sign with a brand new key?

Reply:
Done!